core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContext.java

```diff
@@ -563,9 +562,12 @@ protected RequestProcessorRegistry buildRequestProcessorRegistry() {
     return new RequestProcessorRegistry(logPrefix, processors.toArray(new RequestProcessor[0]));
   }
 
-  protected CodecRegistry buildCodecRegistry(String logPrefix, List<TypeCodec<?>> codecs) {
-    MutableCodecRegistry registry = new DefaultCodecRegistry(logPrefix);
-    registry.register(codecs);
+  protected CodecRegistry getCodecRegistry(ProgrammaticArguments arguments) {
+    MutableCodecRegistry registry = (MutableCodecRegistry) arguments.getCodecRegistry();
```

Comment:
So, that `MutableCodecRegistry` type is a bit awkward, it only exists for backward compatibility issue (maybe you've seen that in the javadocs already). While all registry implementations *should* extend it, it's possible for someone to pass an implementation that doesn't, we can't risk an exception here.

One option is to do an `instanceof` check, and skip the registration of the codecs if it fails. We should also log a warning, because if someone provided codecs they expect them to work.

Another is to require the mutable type directly at build time: `SessionBuilder.withCodecRegistry(MutableCodecRegistry)`.

I prefer the latter, WDYT?

Reply:
The former option feels more correct, while the second adds less clutter to the code. I find it hard to decide, so I'll defer to you and go with MutableCodecRegistry all the way.

Reply:
Sounds good. Another way to put it is that MutableCodecRegistry is the type that should have been exposed publicly from the start.

Reply:
All the changes should be there, thanks again for reviewing!